.github/workflows/ci.yml

```
@@ -39,20 +39,31 @@ jobs:
       - name: Run ShellCheck
         uses: ludeeus/action-shellcheck@master
         with:
-            ignore_paths: ./.devcontainer
+          ignore_paths: ./.devcontainer
 
   test-install:
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        shell: [sh, bash, zsh]
+        shell: [sh, bash, zsh, fish]
```

```suggestion
        include:
          - os: ubuntu-latest
            install-approach: apt
          - os: ubuntu-latest
            install-approach: sh
          - os: macos-latest
            install-approach: brew
```

Etcetera. You should consider using [includes](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrixinclude) here so you don't have a bunch of do-nothings like "windows fish".